fix: update auth_url validation method

thargosu opened the pull request:
It seems netflix now returns tokens which can exceed 42 characters in length.
This code updates the validation mecanism to take this into account. fix #1740

### Check if this PR fulfills these requirements:
* [x] My changes respect the [Kodi add-on development rules](https://kodi.wiki/view/Add-on_rules)
* [x] I have read the [**CONTRIBUTING**](../../master/Contributing.md) document.
* [x] I made sure there wasn't another [Pull Request](../../../pulls) opened for the same update/change
* [x] I have successfully tested my changes locally

#### Types of changes
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Feature change (non-breaking change which changes behaviour of an existing functionality)
- [ ] Improvement (non-breaking change which improves functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor (non-breaking performance or readability improvements)

### Description
<!--- Motivation and a possible detailed explanation of the changes -->
<!--- Put your text below this line -->

### In case of Feature change / Breaking change:
#### Describe the current behavior
<!--- Put your text below this line -->

#### Describe the new behavior
<!--- Put your text below this line -->

### Screenshots (if appropriate):
<!--- Add some screenshots if they can be useful to show the differences between before and after the changes -->

martinbilek commented:
This works for me. Good work!

thargosu commented on resources/lib/utils/website.py:
```diff
@@ -240,7 +240,8 @@ def extract_api_data(react_context, debug_log=True):
     """Raise an exception if user_data does not contain a valid authURL"""
-    if len(user_data.get('auth_url', '')) != 42:
+    auth_url = user_data.get('auth_url', '')
+    if not auth_url.startswith('c1.') or len(auth_url) < 10:
```
Good question—thanks for pointing that out. Initially, I assumed the auth_url would consistently start with "c1." based on my test data, but it seems some users have encountered different formats. It might make more sense to rely on the length instead, though I'm not yet certain if it's consistently 77 characters. I’ll look into it further and adjust the validation logic as needed.

jankofron replied:
Can confirm this. Starting two weeks ago (Netflix singed out), I could not login anymore (invalid URL...), the length of the response is 77 also in my case. A similar fix (just commented out the length check) worked for me to log in again, but all videos I tried to play stop after 2 minutes (just audio continues). So, still, this does not solve my problem. Could not get any further, now, sadly...

thargosu replied:
After further testing I can see that the auth token is always starting with `c1.`. Then there is a timestamp indicating the time at which the token was produced. Finaly there is a 61 characters long base64 encoded string representing what seems to be protobuf data from what I could understand.
The whole string seems to always be 77 characters long.

Example:

`c1.1747244436955.AgiMlOvcAxIggrNFSlPwSU6ukluBMQ7v1QqS77W2F6XXXXXXXXXXXXXXXX==`

`c1` : could be some kind of version or context.
`1747244436955`: GMT: Wednesday, May 14, 2025 17:40:36.955 
`AgiMlOvcAxIggrNFSlPwSU6ukluBMQ7v1QqS77W2F6XXXXXXXXXXXXXXXX==` : base64 string

```
$ echo AgiMlOvcAxIggrNFSlPwSU6ukluBMQ7v1QqS77W2F6XXXXXXXXXXXXXXXX== | base64 -d > /tmp/token.bin
$ tail -c +2 /tmp/token.bin | protoc --decode_raw
1: 1000000012
2: "\202\263EJS\360IN\256\222 .......... \260Y\227\362\324}\345r\273"
3: 2
```

There are other issues after that in the logs:
`2025-05-14 19:40:37.878 T:1016    error <general>: [plugin.video.netflix (0)] It was not possible to extract the client version!`

I think this pull request might not be enough as of now. I'll try to see if I can fix the client version extraction issue.

thargosu replied:
Ok so I checked the client version extraction and it turns out the regex line 96 of website.py doesn't fit anymore.
The asset_core now looks like this:
```
https://assets.nflxext.com/player/html/ffe/cadmium-playercore-6.0049.787.911.js?testid=70117
```
For some reasons there is now an additional string after the `.js`.

The regex line 96 is like this:
```
result = search(r'-([0-9\.]+)\.js$', api_data.pop('asset_core'))
```
Hence it fails.

I'll try to update the PR to fix this as well.

thargosu commented:
Hi,
After several more tests, I feel confident that the token size is now 77. I have updated the pull request accordingly.
I have also fixed the client version extraction line 96.

I'm happy to have your thoughts on this proposal.
Best regards

catch56 commented:
Latest version of the PR works for me. Left one review comment.

thargosu commented on resources/lib/utils/website.py:
```diff
@@ -240,7 +240,8 @@ def extract_api_data(react_context, debug_log=True):
     """Raise an exception if user_data does not contain a valid authURL"""
-    if len(user_data.get('auth_url', '')) != 42:
+    auth_url = user_data.get('auth_url', '')
+    if not auth_url.startswith('c1.') or len(auth_url) != 77:
```
I agree. It might be overkill and have side effects.
I'll push another version without the check on the c1 string.

CastagnaIT reviewed (approved):
thanks although i can't test locally at the moment the changes seem reasonable
